pepper-apis/src/main/java/org/broadinstitute/ddp/route/ReceiveDsmNotificationRoute.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.ddp.route;
+
+import org.apache.http.HttpStatus;
+import org.broadinstitute.ddp.constants.ErrorCodes;
+import org.broadinstitute.ddp.constants.RouteConstants.PathParam;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.JdbiUserStudyEnrollment;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.json.dsm.DsmNotificationPayload;
+import org.broadinstitute.ddp.json.errors.ApiError;
+import org.broadinstitute.ddp.model.activity.types.DsmNotificationEventType;
+import org.broadinstitute.ddp.model.event.DsmNotificationSignal;
+import org.broadinstitute.ddp.model.user.EnrollmentStatusType;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.service.EventService;
+import org.broadinstitute.ddp.util.ResponseUtil;
+import org.broadinstitute.ddp.util.RouteUtil;
+import org.broadinstitute.ddp.util.ValidatedJsonInputRoute;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+/**
+ * A hook called by DSM at a certain stage, triggers the notification sending when invoked
+ * Given a user GUID, a study GUID and a notification event type, looks for a notification template
+ * and queues an event which is picked up by Housekeeping later and results in sending the notification
+ * Returns:
+ *    200 OK - the notification event was queued or no event configuration exists
+ *    404 Not Found - no study, user or notification event type found
+ */
+public class ReceiveDsmNotificationRoute extends ValidatedJsonInputRoute<DsmNotificationPayload> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ReceiveDsmNotificationRoute.class);
+
+    @Override
+    protected int getValidationErrorStatus() {
+        return HttpStatus.SC_BAD_REQUEST;
+    }
+
+    @Override
+    public Object handle(Request request, Response response, DsmNotificationPayload payload) {
```

Comment:
From DSM's perspective, the endpoint should be the same. I have updated some of the error responses and messages, but I don't think that will affect DSM. CC @pegahtah for good measure.